tests/model/test_cycle_data.py

```diff
@@ -412,6 +417,77 @@ def test_irrelevant_repos_data_removal(
     ).exists()
 
 
+def test_signature_remover(test_perf_signature, test_perf_signature_2, test_perf_data):
+    cycler = PerfherderCycler(chunk_size=100, sleep_time=0)
+    expired_timestamp = cycler.max_timestamp
+    test_perf_signature_2.last_updated = expired_timestamp
+    test_perf_signature_2.save()
+
+    assert len(PerformanceSignature.objects.all()) == 2
+
+    call_command('cycle_data', 'from:perfherder')
+
+    assert taskcluster.Notify().ping.called_once
+    assert taskcluster.Notify().email.call_count == 2
+    assert len(PerformanceSignature.objects.all()) == 1
+    assert PerformanceSignature.objects.first() == test_perf_signature
+
+
+def test_signature_remover_when_notify_service_is_down(
+    test_perf_signature, test_perf_signature_2, test_perf_data
+):
+    taskcluster.Notify().ping.side_effect = Exception('Email Service is down.')
+
+    call_command('cycle_data', 'from:perfherder')
+
+    assert taskcluster.Notify().ping.called_once
+    assert not taskcluster.Notify().email.called
+    assert len(PerformanceSignature.objects.all()) == 2
+
+
+@pytest.mark.parametrize('total_signatures', [3, 4, 8, 10])
+def test_nr_emails_sent(test_perf_signature, total_signatures):
```

Comment:
I still see mentionings of "nr" throughout this test module. Please address all of these.